src/wallet/wallet.h

```diff
@@ -647,12 +648,66 @@ class LelantusJoinSplitBuilder;
 //static boost::signals2::signal<void (CWallet *wallet)> UnlockWallet;
 extern boost::signals2::signal<void (CWallet *wallet)> UnlockWallet;
 
+class CTransparentTxout;
+enum AbstractTxoutType {
+    Transparent
+};
+
+class CTransparentTxout {
+public:
+    static bool IsTransparentTxout(const CTxOut& txout);
+
+    CTransparentTxout() = default;
+    CTransparentTxout(COutPoint outpoint, CTxOut txout): outpoint(outpoint), txout(txout), _isMockup(true) {}
+    CTransparentTxout(const CWallet* wallet, COutPoint outpoint, CTxOut txout): wallet(wallet), outpoint(outpoint), txout(txout) {};
+
+    uint256 GetHash() const;
+    COutPoint GetOutpoint() const;
+    CAmount GetValue() const;
+    CScript GetScriptPubkey() const;
+    size_t GetMarginalSpendSize(std::vector<CTransparentTxout>& previousInputs) const;
+    bool IsMine(const CCoinControl* coinControl) const;
+    bool IsSpendable() const;
+    bool IsLocked() const;
+    bool IsAbandoned() const;
+    bool IsCoinTypeCompatible(const CCoinControl* coinControl) const;
+    bool IsLLMQInstantSendLocked() const;
+    bool IsCoinBase() const;
+    bool IsFromMe() const;
+    unsigned int GetDepthInMainChain() const;
+    unsigned int GetDepthInMempool() const;
+
+private:
+    const CWallet* wallet = nullptr;
+    COutPoint outpoint;
+    CTxOut txout;
+    bool _isMockup = false;
+
+public:
+    bool _mockupIsMine = false;
+    bool _mockupIsMineWatchOnly = false;
+    bool _mockupIsSpent = false;
+    bool _mockupIsAbandoned = false;
+    bool _mockupIsLocked = false;
+    bool _mockupIsLLMQInstantSendLocked = false;
+    bool _mockupIsCoinBase = false;
+    unsigned int _mockupDepthInMainChain = 0;
+};
+
 /**
  * A CWallet is an extension of a keystore, which also maintains a set of transactions and balances,
  * and provides the ability to create new transactions.
  */
 class CWallet : public CCryptoKeyStore, public CValidationInterface
 {
+public:
+    /**
+     * Used to keep track of spent outpoints, and
+     * detect and report conflicts (double-spends or
+     * mutated transactions where the mutant gets mined).
+     */
+    typedef std::multimap<COutPoint, uint256> TxSpends;
```

Comment:
it's only a type but I can't find its usage of TxSpends outside of the CWallet class, why change it to public then?